### PR TITLE
fix #167: [FR] Redirect conjuged verbs to their infinitive form

### LINKED ISSUE #167: [FR] Redirect conjuged verbs to their infinitive form
As [requested](http://www.tiger-222.fr/?d=2020/04/17/22/14/21-un-dictionnaire-alternatif-et-complet-pour-votre-kobo#id2764d8) it would be cool to have conjuged verbs redirecting to their infinitive form instead of nothing.

I already tried some things, but without success. I *think* we could make use of [variants](https://pgaskin.net/dictutil/dicthtml/format.html#html-dicthtml), but it is not clear yet how to do that.

### PR DESCRIPTION
**Pull request by lasconic**
- [x] Fixes #167 
- [x] Tests added/updated.
- [x] `./check.sh` run and eventual issues fixed.
- [x] Updated CONTRIBUTORS.md, if not already done in the past.

### CODE REVIEW

File: scripts/convert.py [hunk 1]
```diff
@@ -53,6 +54,16 @@ def make_groups(words: Words) -> Groups:
     return groups
 
 
+def make_variants(words: Words) -> Variants:
```
Comment (BoboTiG) [inline scripts/convert.py:57]: Could you add a docstring please?

File: scripts/get.py [hunk 2]
```diff
@@ -159,6 +159,10 @@ def find_section_definitions(
     """Find definitions from the given *section*, with eventual sub-definitions."""
     definitions: List[Definitions] = []
 
+    # do not look for definions in french verb form section
+    if locale == "fr" and section.title.strip() == "{{S|verbe|fr|flexion}}":
```
Comment (BoboTiG) [inline scripts/get.py:163]: For multiple flexions, we could do it that way:
```suggestion
    if locale == "fr" and section.title.strip().startswith("{{S|verbe|fr|flexion"):
```

File: scripts/get.py [hunk 3]
```diff
@@ -395,7 +399,17 @@ def parse_word(word: str, code: str, locale: str, force: bool = False) -> Word:
         pron = find_pronunciation(code, pronunciation[locale])
         nature = find_genre(code, genre[locale])
 
-    return Word(pron, nature, etymology, definitions)
+    # find if  variant and delete unwanted definitions
+    variant = ""
+    if locale == "fr" and not definitions:  # Pure verb form, no definition
+        for title, s in parsed_sections.items():
+            if "{{S|verbe|fr|flexion}}" == title:
```
Comment (BoboTiG) [inline scripts/get.py:406]:
```suggestion
            if title.startswith("{{S|verbe|fr|flexion"):
```

File: scripts/lang/fr/__init__.py [hunk 4]
```diff
@@ -106,6 +106,7 @@
     "{{S|verbe|fr|loc",
     "{{S|verbe|fr|num",
     "{{S|verbe|fr}",
+    "{{S|verbe|fr|flexion}}",
```
Comment (BoboTiG) [inline scripts/lang/fr/__init__.py:109]:
```suggestion
    "{{S|verbe|fr|flexion",
```

File: scripts/lang/fr/__init__.py [hunk 5]
```diff
@@ -608,6 +609,8 @@
     "wsp": "parts[2] if len(parts) > 2 else parts[1]",
     # {{WSP|Panthera leo}}
     "WSP": "term(parts[1])",
+    # {fr-verbe-flexion|colliger|ind.i.3s=oui}}
+    "fr-verbe-flexion": "parts[1] if parts[1] != 'grp=3' else parts[2]",
```
Comment (BoboTiG) [inline scripts/lang/fr/__init__.py:613]: Could you keep the dict sorted?

File: scripts/stubs.py [hunk 6]
```diff
@@ -4,5 +4,5 @@
 
 Definitions = Union[str, Tuple[str, ...]]
 Parts = Tuple[str, ...]
-Word = namedtuple("Word", "pronunciation, genre, etymology, definitions")
+Word = namedtuple("Word", "pronunciation, genre, etymology, definitions, variant")
```
Comment (BoboTiG) [inline scripts/stubs.py:7]: As there may be several variants, should we use plural here? If so, apply changes everywhere :)

File: tests/test_fr.py [hunk 7]
```diff
@@ -380,6 +427,8 @@ def test_parse_word(word, pronunciation, genre, etymology, definitions, page):
             "{{siècle|lang=fr|XI}} Du {{étyl|frm|fr|chambre}}<ref>{{R:DMF}}</ref>, de l’{{étyl|fro|fr|chambre}}, {{lien|''cambre''|fro}}, {{lien|''cambra''|fro}}<ref>{{R:DÉCT}}</ref>{{,}}{{R|TLFi}}, du {{étyl|bas latin|fr|mot=camera|sens=pièce, chambre}}{{R|TLFi}}, du {{étyl|latin classique|fr|mot=camera}}{{R|TLFi}}, {{lien|''camara''|sens=voute, plafond vouté|la}}, du {{étyl|grc|fr|mot=καμάρα|tr=kamárā|sens=voute, lieu couvert par une construction}}{{R|TLFi}}.",  # noqa
             "<i>(XI<sup>e</sup> siècle)</i> Du moyen français <i>chambre</i>, de l’ancien français <i>chambre</i>, <i>cambre</i>, <i>cambra</i>, du bas latin <i>camera</i> (« pièce, chambre »), du latin classique <i>camera</i>, <i>camara</i> (« voute, plafond vouté »), du grec ancien καμάρα, <i>kamárā</i> (« voute, lieu couvert par une construction »).",  # noqa
         ),
+        ("{{fr-verbe-flexion|colliger|ind.i.3s=oui}}", "colliger"),
+        ("{{fr-verbe-flexion|grp=3|couvrir|ind.i.3s=oui}}", "couvrir"),
```
Comment (BoboTiG) [inline tests/test_fr.py:431]: Could you keep the list sorted here too?

File: scripts/get.py [hunk 8]
```diff
@@ -395,7 +399,17 @@ def parse_word(word: str, code: str, locale: str, force: bool = False) -> Word:
         pron = find_pronunciation(code, pronunciation[locale])
         nature = find_genre(code, genre[locale])
 
-    return Word(pron, nature, etymology, definitions)
+    # find if  variant and delete unwanted definitions
```
Comment (BoboTiG) [inline scripts/get.py:402]:
```suggestion
    # find if variant and delete unwanted definitions
```

File: scripts/get.py [hunk 9]
```diff
@@ -159,6 +159,10 @@ def find_section_definitions(
     """Find definitions from the given *section*, with eventual sub-definitions."""
     definitions: List[Definitions] = []
 
+    # do not look for definions in french verb form section
```
Comment (BoboTiG) [inline scripts/get.py:162]:
```suggestion
    # do not look for definitions in french verb form section
```

File: scripts/stubs.py (hunk 6 above)
Comment (lasconic) [reply scripts/stubs.py:7]: There might be several variant indeed ("mirent" -> mirer / mettre) but in the end we need to choose one to add the word in the HTML file... So not sure it's worth making it plural. It's a string.

Comment (BoboTiG) [reply scripts/stubs.py:7]: That's my point: variants could be a list of strings and we would have to handle "mirer" and "mettre". Maybe useless for now :)

**Comment by lasconic** [comment]
I addressed most of the remarks. I'm not sure about "variant"/"variants". As I said, it's a single variant, we just need to choose. I chose the first one and broke the loop: https://github.com/BoboTiG/ebook-reader-dict/pull/191/files#diff-d00330496418af6b203d88de9efa2fffe42ccc66921d336b7455b852e77c0ca8R411

**Comment by BoboTiG** [comment]
Great!
I merged to have the new words in the next auto-update. Let's handle next fixes/improvements in specific PRs.

Thanks for the work :)

**Pull request merged**